### ISSUE TITLE
`thamos images` errors

### ISSUE THREAD

**Issue by goern** [thoth-station/thamos#1184]
`thamos images` throws an error: 

/kind bug

**Comment by codificat** [comment]
Related: #1109

**Comment by goern** [comment]
delivering a 404 sounds reasonable when no images are available, or when requesting page 5 with per_page 10 and a total of 8 images available. 404 is the server signaling, that the requrested data could not be found, giving 200 with an empty list if a a incorrect behavior, isnt it?

**Comment by codificat** [comment]
> delivering a 404 sounds reasonable when no images are available

Maybe. But:
- a 404 would also be returned if you try to access a non-existing API endpoint. Same code for different meanings.
- returning a 200 with an empty result sounds also reasonable to me: a way of saying "I found 0 occurrences of this", which is not necessarily an error.

> or when requesting page 5 with per_page 10 and a total of 8 images available. 404 is the server signaling, that the requrested data could not be found, 

That might better be answered by a 400 (bad request) instead?  Or even a 200 with empty results (meaning "this is what I found: nothing").

Note that the API response includes a total item count and a page count in the response headers; these complement an empty 200 response nicely I think.

> giving 200 with an empty list if a a incorrect behavior, isnt it?

IMO I think it is a better behavior when appropriate... but I can buy the 404 too, no too-hard feelings.
